packages/api/src/domain/medical/address.ts

```diff
@@ -7,8 +13,28 @@ export type Address = {
   state: USState;
   zip: string;
   country?: string;
+  coordinates?: Coordinates;
 };
 
+/**
+ * This function takes two lists of addresses, and combines them into a single list of addresses. It prioritizes the addresses with coordinates and removes duplicates.
+ * @param addressList1 - the first list of addresses.
+ * @param addressList2 - the second list of addresses.
+ * @returns - a list of addresses.
+ */
+export function combineAddresses(addressList1: Address[], addressList2: Address[]): Address[] {
+  const combined = [...addressList1, ...addressList2];
+
+  const compareAddresses = (a: Address, b: Address): number => {
+    const aHasCoordinates = a.coordinates ? 1 : 0;
+    const bHasCoordinates = b.coordinates ? 1 : 0;
+    return bHasCoordinates - aHasCoordinates;
+  };
+  combined.sort(compareAddresses);
+
+  return uniqBy(combined, a => `${a.addressLine1}-${a.addressLine2}-${a.city}-${a.state}-${a.zip}`);
```

Comment:
could make simpler with this:
`return uniqBy(combined, JSON.stringify);`

Reply:
This looks very clean, but I read that this might not be as robust, given that string representations of an object will be considered different simply if the attributes are in a different order.

Reply:
ah ok sounds good to me then